src/App.tsx

```diff
@@ -15,48 +16,110 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  name = 'name',
+  length = 'length',
+}
+
+const REVERSE = 'reverse';
+
+export function getPreparedGoods(
+  goods: string[],
+  howSort: SortType | '',
+  reverse = '',
+) {
+  const preparedGoods = [...goods];
+
+  if (howSort) {
+    preparedGoods.sort((good1, good2) => {
+      switch (howSort) {
+        case SortType.name:
+          return good1.localeCompare(good2);
+
+        case SortType.length:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverse) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortBy, setSortBy] = useState<SortType | ''>('');
```

Comment:
you can describe in `SortType` than name can be `'name' | ''` and then you dont need to use empty string inside generic